Modix.Data/Migrations/20200125175253_AddInfractionMessagePromotionCampaignNavigationToGuildUser.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+namespace Modix.Data.Migrations
+{
+    public partial class AddInfractionMessagePromotionCampaignNavigationToGuildUser : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateIndex(
+                name: "IX_Messages_ChannelId",
+                table: "Messages",
+                column: "ChannelId");
+
+            migrationBuilder.AddForeignKey(
+                name: "FK_Messages_GuildChannels_ChannelId",
+                table: "Messages",
+                column: "ChannelId",
+                principalTable: "GuildChannels",
+                principalColumn: "ChannelId",
+                onDelete: ReferentialAction.Cascade);
+
+            migrationBuilder.AddForeignKey(
+                name: "FK_Messages_GuildUsers_GuildId_AuthorId",
+                table: "Messages",
+                columns: new[] { "GuildId", "AuthorId" },
+                principalTable: "GuildUsers",
+                principalColumns: new[] { "GuildId", "UserId" },
+                onDelete: ReferentialAction.Cascade);
```

Comment:
Does having this level of referential integrity between the messages and `GuildUsers` table buy us anything? This would make inserting a message fail if, for some odd reason, the author of the  message didn't exist in the `GuildUsers` table - right?

Also, is this (and the foreign key above it) safe, given that these columns are different data types?

Reply:
There is no benefit here other than integrity - this was automatically generated by Entity Framework after the missing relationships were added.

We _could_ forego the suggestion here, seeing as the messages table is high throughput and this could hurt performance. I'm not sure what the implications are in a Postgres environment.